sql/drop.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: XisiHuang (cockhuangxh@163.com)
+
+package sql
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/keys"
+	"github.com/cockroachdb/cockroach/sql/parser"
+)
+
+// DropTable drops a table.
+func (p *planner) DropTable(n *parser.DropTable) (planNode, error) {
+	// TODO(XisiHuang): should do truncate and delete descriptor in
+	// the same txn
+	for i, tableQualifiedName := range n.Names {
+		if err := p.normalizeTableName(tableQualifiedName); err != nil {
+			return nil, err
+		}
+
+		dbDesc, err := p.getDatabaseDesc(tableQualifiedName.Database())
+		if err != nil {
+			return nil, err
+		}
+
+		if !dbDesc.HasPrivilege(p.user, parser.PrivilegeWrite) {
```

Comment:
actually, can you remove it entirely? This should be a table-level privilege (don't ask, that's just the way it is).
I'm in the middle of implementing table-level privileges and corresponding tests.